app/models/miq_provision/options_helper.rb

```diff
@@ -25,12 +25,13 @@ def create_template?
 
   def get_source
     source_id = get_option(:src_vm_id)
-    source = VmOrTemplate.find_by_id(source_id)
+    source_type = get_option(:src_type)
+    source = MiqProvisionSource.get_provisioning_request_source(source_id, source_type)
```

Comment:
I don't know why, but all over the place in the provisioning/miq request code, the source relation has been overloaded so it can't be trusted to just be the model that was signed to it. I just continued to work around that issue in the same way the existing code did.